kevm-pyk/src/kevm_pyk/solc_to_k.py

```diff
@@ -42,20 +42,26 @@ def __init__(self, name: str, id: int, abi: Dict, contract_name: str, sort: KSor
             # TODO: Check that we're handling all state mutability cases
             self.payable = abi['stateMutability'] == 'payable'
 
+        @property
+        def signature(self) -> str:
+            arg_list = ','.join(self.arg_types)
+            return f'{self.name}({arg_list})'
+
         @property
         def selector_alias_rule(self) -> KRule:
-            return KRule(KRewrite(KEVM.abi_selector(self.name), intToken(self.id)))
+            return KRule(KRewrite(KEVM.abi_selector(self.signature), intToken(self.id)))
 
         @property
         def production(self) -> KProduction:
             input_nonterminals = (KNonTerminal(_evm_base_sort(input_type)) for input_type in self.arg_types)
             items_before: List[KProductionItem] = [KTerminal(self.name), KTerminal('(')]
             items_args: List[KProductionItem] = list(intersperse(input_nonterminals, KTerminal(',')))
             items_after: List[KProductionItem] = [KTerminal(')')]
+            args_list = '_'.join([_evm_base_sort(input_type).name for input_type in self.arg_types])
             return KProduction(
                 self.sort,
                 items_before + items_args + items_after,
-                klabel=KLabel(f'method_{self.contract_name}_{self.name}'),
+                klabel=KLabel(f'method_{self.contract_name}_{self.name}_{args_list}'),
```

Comment:
Let's factor out the `klabel` as a property of `Method`, and use that property here, and in `rule`, and in `application`.